OBSDOCS-1857: Fix Loki retention module and remove the duplicate module (#95075)
openshift/openshift-docs: modules/logging-loki-retention.adoc

// Module included in the following assemblies:
//
// * observability/logging/log_storage/cluster-logging-loki.adoc
// * configuring/configuring-the-log-store.adoc

:_mod-docs-content-type: PROCEDURE
[id="logging-loki-retention_{context}"]
= Enabling stream-based retention with Loki

You can configure retention policies based on log streams. You can set retention rules globally, per-tenant, or both. If you configure both, tenant rules apply before global rules.

include::snippets/logging-retention-period-snip.adoc[]

[NOTE]
====

> * Although logging version 5.9 and later supports schema `v12`, `v13` is recommended because it is the schema version to be supported going forward.

Author: Added missing context to why v13 is recommended, based on the existing documentation: https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/logging/cluster-logging-upgrading?utm_source=chatgpt.com#logging-upgrading-loki-schema_cluster-logging-upgrading

// Module included in the following assemblies:
//
// * observability/logging/log_storage/cluster-logging-loki.adoc
// * configuring/configuring-the-log-store.adoc

:_mod-docs-content-type: PROCEDURE
[id="logging-loki-retention_{context}"]
= Enabling stream-based retention with Loki

> You can configure retention policies based on log streams. You can set retention rules globally, per-tenant, or both. If you configure both, tenant rules apply before global rules.

Author: Keeping per-tenant as that is the version seen in other parts of OCP documentation.

include::snippets/logging-retention-period-snip.adoc[]

[NOTE]
====

> * Although logging version 5.9 and later supports schema `v12`, schema `v13` is recommended because it is the schema version to be supported going forward.

Reviewer: ```suggestion
* Although logging version 5.9 and later supports schema `v12`, schema `v13` is recommended for future compatibility.
```
Or something? Attempting to reduce wordiness.

Author: Great improvement, thanks!

> * For cost-effective log pruning, configure retention policies directly on the object storage provider. Using the storage provider's lifecycle management ensures automatic deletion of old logs and avoids extra processing from Loki and delete requests to s3.

Reviewer 2: ```suggestion
* For cost-effective log pruning, configure retention policies directly on the object storage provider. Using the storage provider's lifecycle management ensures automatic deletion of old logs. This also avoids extra processing from Loki and delete requests to S3.
```

Splitting the sentence to make it simpler.

> * For cost-effective log pruning, configure retention policies directly on the object storage provider. Using the storage provider's lifecycle management ensures automatic deletion of old logs. This also avoids extra processing from Loki and delete requests to S3.

Reviewer: ```suggestion
* For cost-effective log pruning, configure retention policies directly on the object storage provider. Use the lifecycle management features of the storage provider to ensure automatic deletion of old logs. This practice also avoids extra processing from Loki and delete requests to S3.
```

* General IBM Style Guide ⚠️ against possessives.
* Ambiguous pronoun clarification

+

> If the object storage does not support lifecycle policies, you must configure LokiStack to enforce retention internally. The supported retention period is up to 30 days.

Reviewer 2: ```suggestion
If the object storage does not support lifecycle policies, you must configure LokiStack to enforce retention internally. LokiStack supports a retention period of up to 30 days.
```

====

.Prerequisites 

* You have administrator permissions.
* You have installed the {loki-op}.
* You have installed the {oc-first}.

.Procedure

> . To enable stream-based retention, create a `LokiStack` CR and save it as a YAML file. In this example, it is called `lokistack.yaml`.

Reviewer: ```suggestion
. To enable stream-based retention, create a `LokiStack` CR and save it as a YAML file. In the following example, it is called `lokistack.yaml`.
```

  storage:
    schemas:
    - effectiveDate: "2020-10-11"
      version: v13
    secret:
      name: logging-loki-s3
      type: s3
  storageClassName: gp3-csi
  tenants:
    mode: openshift-logging
----

> <1> Set the retention policy for all log streams. This does not impact the retention period for stored logs in object storage.

Reviewer: ```suggestion
<1> Set the retention policy for all log streams. This policy does not impact the retention period for stored logs in object storage.
```

Ambiguity avoidance.
